project_argparser.py

```diff
@@ -1,38 +1,146 @@
 import argparse
+import os.path
+import csv
+import ast
 
-#Creating the parser
-parser = argparse.ArgumentParser(description="Electric car racing simulation")
-
-#This class is meant to make it easier to enter two correct inputs and check against them
-#Preferably the name you give to an object of this class would be the same (or as close to the same without conflict) as the name of the argument created within it
-#...to allow for easily calling arg_check() (ex:logging_arg.arg_check(args.logging))
 class SingleArg:
 
     def __init__(self, parser, key, lng_key, help_msg, on_msg, off_msg):
         #Adds an argument with the key (ex: -l), name (ex: --logging), and help message to be displayed when entering -h
-        parser.add_argument(key, lng_key, type=str, help=help_msg)
+        parser.add_argument(key, lng_key, type=str, help=help_msg, default=off_msg)
         #This sets the strings for which input will be checked against in arg_check()
         self.on_msg = on_msg
         self.off_msg = off_msg
     
+    #argument checking for on/off or other binary arguments
     def arg_check(self, input):
         if (input == self.on_msg):
             return True
         elif (input == self.off_msg):
             return False
         else:
-            return False
-            #raise argparse.ArgumentTypeError('Invalid input — use -h for more information on arguments.') 
+            raise argparse.ArgumentTypeError('Invalid input — use -h for more information on arguments.') 
+    
+    #opens csv and creates dict with keys corresponding to the headers of the fastsim car csv file format
+    def open_car_dict(self, input):
+        if input == None:
+            input = self.off_msg
+        if not os.path.exists(input):
+            raise argparse.ArgumentTypeError('The file %s is not in the working directory' % input)
+        else:
+            with open(input, newline='') as csv_file:
+                csv_data = list(csv.reader(csv_file))
+        car_dict = dict()
+        for i in range(len(csv_data[0])):
+            car_dict[csv_data[0][i]] = eval_type(csv_data[1][i]) 
 
-    #My thought is — in the future more methods like arg_check() can be implemented to allow for checking on non-binary arguments such as file names, etc.
+        #Summing total car mass -- function borrowed from fastsim and adapted
 
+        """Calculate total vehicle mass.  Sum up component masses if 
+        positive real number is not specified for self.vehOverrideKg"""
+        ess_mass_kg = 0
+        mc_mass_kg = 0
+        fc_mass_kg = 0
+        fs_mass_kg = 0
+        if (isinstance(car_dict["vehOverrideKg"], int)):
```

Comment:
#TODO clean up and possibly import fastsim functions to take care of this kind of thing